### PR TITLE
Multiple files

### PR DESCRIPTION
**Pull request by shikaan**
## General info

**Type of issue:** FEATURE

## Problem
Currently Titef only allows using one source file at a time. This option is meant to add the facility of testing a whole folder, possibly filtering elements by extension

## Solution
A new command line helper has been added which is responsible of taking a string of comma separated extensions and list all the files in a subdirectory having that given extension

## Comments
If this is deemed as enough, documentation also should be updated

### CODE REVIEW

File: lib/cli/index.js [hunk 1]
```diff
@@ -1,37 +1,85 @@
...
+  VERSION: 'version',
+  EXTENSIONS: 'extensions',
+  DIRECTORY: 'directory',
+};
 
-const showHelp = nonTrivialArguments.find(item => item === '-h' || item === '--help');
-const showVersion = nonTrivialArguments.find(item => item === '-v' || item === '--version');
+function getFlagValue(argv, flag, shortcut) {
+  return shortcut
+    ? !!argv.find(item => item === flag || item === shortcut)
```
Comment (shikaan) [inline lib/cli/index.js]: Codacy Issue found: [Expected parentheses around arrow function argument.](https://app.codacy.com/manual/spagnolo.manu/titef/pullRequest?prid=5052259)

**Comment by shikaan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/spagnolo.manu/titef/pullRequest?prid=5052259&bid=16633666)

File: lib/cli/index.js [hunk 2]
```diff
@@ -1,37 +1,85 @@
...
+    : !!argv.find(item => item === flag);
+}
+
+function getArgumentValue(argv, option, shortcut) {
+  const optionArgvItemIndex = argv.findIndex(item => item.includes(option) || item.includes(shortcut));
```
Comment (shikaan) [inline lib/cli/index.js]: Codacy Issue found: [This line has a length of 103. Maximum allowed is 100.](https://app.codacy.com/manual/spagnolo.manu/titef/pullRequest?prid=5052259)

File: lib/cli/get-files.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    const itemPath = join(rootDirectory, item);
+
+    const stats = statSync(itemPath);
+
+    if (stats.isDirectory()) {
+      const childrenFilteredFiles = getFiles(itemPath, filters);
+
+      paths.push(...childrenFilteredFiles);
+    } else {
+      const fileHasAllowedExtension = extensions.some(extension => itemPath.endsWith(extension));
```
Comment (shikaan) [inline lib/cli/get-files.js]: Codacy Issue found: [Expected parentheses around arrow function argument.](https://app.codacy.com/manual/spagnolo.manu/titef/pullRequest?prid=5052259)

File: lib/cli/index.js [hunk 4]
```diff
@@ -1,37 +1,85 @@
...
+  EXTENSIONS: 'extensions',
+  DIRECTORY: 'directory',
+};
 
-const showHelp = nonTrivialArguments.find(item => item === '-h' || item === '--help');
-const showVersion = nonTrivialArguments.find(item => item === '-v' || item === '--version');
+function getFlagValue(argv, flag, shortcut) {
+  return shortcut
+    ? !!argv.find(item => item === flag || item === shortcut)
+    : !!argv.find(item => item === flag);
```
Comment (shikaan) [inline lib/cli/index.js]: Codacy Issue found: [Expected parentheses around arrow function argument.](https://app.codacy.com/manual/spagnolo.manu/titef/pullRequest?prid=5052259)

File: lib/cli/index.js (hunk 2 above)
Comment (shikaan) [inline lib/cli/index.js]: Codacy Issue found: [Expected parentheses around arrow function argument.](https://app.codacy.com/manual/spagnolo.manu/titef/pullRequest?prid=5052259)

**Comment by shikaan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/spagnolo.manu/titef/pullRequest?prid=5052259&bid=16633666)

File: lib/cli/index.js (hunk 2 above)
Comment (shikaan) [inline lib/cli/index.js]: Codacy Issue found: [Expected parentheses around arrow function argument.](https://app.codacy.com/manual/spagnolo.manu/titef/pullRequest?prid=5052259)

Comment (shikaan) [inline lib/cli/index.js]: Codacy Issue found: [This line has a length of 103. Maximum allowed is 100.](https://app.codacy.com/manual/spagnolo.manu/titef/pullRequest?prid=5052259)

File: lib/cli/index.js (hunk 1 above)
Comment (shikaan) [inline lib/cli/index.js]: Codacy Issue found: [Expected parentheses around arrow function argument.](https://app.codacy.com/manual/spagnolo.manu/titef/pullRequest?prid=5052259)

File: lib/cli/get-files.js (hunk 3 above)
Comment (shikaan) [inline lib/cli/get-files.js]: Codacy Issue found: [Expected parentheses around arrow function argument.](https://app.codacy.com/manual/spagnolo.manu/titef/pullRequest?prid=5052259)

File: lib/cli/index.js (hunk 4 above)
Comment (shikaan) [inline lib/cli/index.js]: Codacy Issue found: [Expected parentheses around arrow function argument.](https://app.codacy.com/manual/spagnolo.manu/titef/pullRequest?prid=5052259)

**Comment by shikaan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 10
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/spagnolo.manu/titef/pullRequest?prid=5052259&bid=16633666)

**Comment by shikaan** [comment]
Related PR for documentation: https://github.com/shikaan/titef-website/pull/1

**Comment by shikaan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 9
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/spagnolo.manu/titef/pullRequest?prid=5052259&bid=16633666)

**Comment by shikaan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 9
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/spagnolo.manu/titef/pullRequest?prid=5052259&bid=16633666)

**Comment by shikaan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 9
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/spagnolo.manu/titef/pullRequest?prid=5052259&bid=16633666)

File: lib/cli/get-files.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
+const { readdirSync, statSync } = require('fs');
+const { join } = require('path');
+
+/**
+ * Returns a list of usable extensions from a comma separated list
+ * @param {string} extensions
+ * @returns {[]string}
```
Comment (lukaszmn) [inline lib/cli/get-files.js]:
```suggestion
 * @returns {string[]}
```
I'm not sure if `[]string` will work, even if - for consistency - better to use `string[]`

Comment (shikaan) [reply lib/cli/get-files.js]: Hey, thanks for this. I think you have picked an old commit, as this should have been already solved.

**Comment by shikaan** [comment]
On another note, as soon as I will have time, I will finish to test these new classes I have added and provide a final commit

File: lib/cli/argv-parser.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
+class ArgvParser {
+  /**
+   * @param {[]string} argv - Argv Array
```
Comment (lukaszmn) [inline lib/cli/argv-parser.js]:
```suggestion
   * @param {string[]} argv - Argv Array
```

File: lib/cli/directory-walker.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  hasFileAllowedExtension(filePath) {
+    return this.options.extensions.some((extension) => filePath.endsWith(extension));
+  }
+
+  /**
+ * Returns the list (possibly) filtred of all the files in the directory
+ *
+ * @param {string} [rootDirectory]
+ *
+ *  @returns {[]string}
```
Comment (lukaszmn) [inline lib/cli/directory-walker.js]:
```suggestion
 *  @returns {string[]}
```

**Comment by shikaan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 9
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/spagnolo.manu/titef/pullRequest?prid=5052259&bid=16633666)

**Pull request merged**